src/app/_api/functions/recalculate-builder-score.ts

```diff
@@ -0,0 +1,40 @@
+"use server";
+
+import { revalidatePath, revalidateTag } from "next/cache";
+import { supabase } from "@/db";
+import { rollbarError } from "@/services/rollbar";
+import { getCurrentUser } from "../data/users";
+import { resyncPassportForUser } from "../external/talent-protocol";
+import { CacheKey } from "../helpers/cache-keys";
+
+export const recalculateBuilderScore = async (): Promise<number> => {
+  const user = await getCurrentUser();
+  if (!user) return 0;
```

Comment:
Improvement: Maybe throw an error here? This error can then be caught on the front end side and a `toast.error` used to inform the user they should try to reconnect.